kp.go

```diff
@@ -370,17 +372,30 @@ func (c *Client) getAccessToken(ctx context.Context) (string, error) {
 
 // getCorrelationId returns the correlation ID value from the given Context, or
 // returns a new UUID if not present
-func (c *Client) getCorrelationId(ctx context.Context) string {
-	if ctx.Value(correlationIdContextKey) != nil {
-		corrId := ctx.Value(correlationIdContextKey).(string)
-		_, err := uuid.Parse(corrId)
-		if err == nil {
-			return corrId
-		}
+func (c *Client) getCorrelationID(ctx context.Context) string {
+	corrID := GetCorrelationID(ctx)
+	if corrID == nil {
+		return uuid.New().String()
 	}
 
-	corrId := uuid.New().String()
-	return corrId
+	return corrID.String()
+}
+
+// WithCorrelationID retuns a context containing the UUID
+func WithCorrelationID(ctx context.Context, uuid *uuid.UUID) context.Context {
```

Comment:
Kinda think it might be best to stick with the name convention above with the auth context:
```suggestion
func NewContextWithCorrelationID(ctx context.Context, uuid *uuid.UUID) context.Context {
```

We are also sort of leaking our internal uuid type, and forcing a certain library/type as input... but that is probably OK considering its probably the de facto UUID library

Reply:
sounds good, making the change.